connectors/src/connectors/zendesk/temporal/activities.ts

```diff
@@ -293,27 +310,102 @@ export async function syncZendeskCategoryActivity({
 /**
  * This activity is responsible for syncing all the articles in a Category.
  * It does not sync the Category, only the Articles.
- *
- * @returns true if the Category was updated, false if it was deleted.
  */
-// eslint-disable-next-line no-empty-pattern
-export async function syncZendeskArticlesActivity({}: {
+export async function syncZendeskArticlesActivity({
+  connectorId,
+  categoryId,
+  currentSyncDateMs,
+  forceResync,
+}: {
   connectorId: ModelId;
   categoryId: number;
   currentSyncDateMs: number;
   forceResync: boolean;
-}) {}
+}) {
+  const connector = await _getZendeskConnectorOrRaise(connectorId);
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const loggerArgs = {
+    workspaceId: dataSourceConfig.workspaceId,
+    connectorId,
+    provider: "zendesk",
+    dataSourceId: dataSourceConfig.dataSourceId,
+  };
+  const configuration = await _getZendeskConfigurationOrRaise(connectorId);
+  const categoryInDb = await _getZendeskCategoryOrRaise({
+    connectorId,
+    categoryId,
+  });
+
+  const accessToken = await getZendeskAccessToken(connector.connectionId);
+  const zendeskApiClient = createZendeskClient({
+    token: accessToken,
+    subdomain: configuration.subdomain,
+  });
+
+  const articles =
+    await zendeskApiClient.helpcenter.articles.listByCategory(categoryId);
+
+  await concurrentExecutor(
+    articles,
+    (article) =>
+      syncArticle({
+        connectorId,
+        brandId: categoryInDb.brandId,
+        categoryId,
+        article,
+        dataSourceConfig,
+        currentSyncDateMs,
+        loggerArgs,
+        forceResync,
+      }),
+    { concurrency: 10 }
+  );
+}
 
 /**
  * This activity is responsible for syncing all the tickets for a Brand.
  */
-// eslint-disable-next-line no-empty-pattern
-export async function syncZendeskTicketsActivity({}: {
+export async function syncZendeskTicketsActivity({
+  connectorId,
+  brandId,
+  currentSyncDateMs,
+  forceResync,
+}: {
   connectorId: ModelId;
   brandId: number;
   currentSyncDateMs: number;
   forceResync: boolean;
-  afterCursor: string | null;
-}): Promise<{ hasMore: boolean; afterCursor: string }> {
-  return { hasMore: false, afterCursor: "" };
+}) {
+  const connector = await _getZendeskConnectorOrRaise(connectorId);
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const loggerArgs = {
+    workspaceId: dataSourceConfig.workspaceId,
+    connectorId,
+    provider: "zendesk",
+    dataSourceId: dataSourceConfig.dataSourceId,
+  };
+  const configuration = await _getZendeskConfigurationOrRaise(connectorId);
+
+  const accessToken = await getZendeskAccessToken(connector.connectionId);
+  const zendeskApiClient = createZendeskClient({
+    token: accessToken,
+    subdomain: configuration.subdomain,
+  });
+  await changeZendeskClientSubdomain({ client: zendeskApiClient, brandId });
```

Comment:
noob question: why are we doing this here?

Reply:
this is a weird feature of zendesk's API, there is no endpoint to select all tickets for a brand, instead you have to change the subdomain for one that is specific to the brand and use the generic `/brands.json` endpoint
Will def add comments as this is very specific